dagql/idtui/frontend_pretty.go

```diff
@@ -742,7 +742,7 @@ func (fe *frontendPretty) update(msg tea.Msg) (*frontendPretty, tea.Cmd) { //nol
 			cmd = &wrapCommand{
 				ExecCommand: cmd,
 				before: func() error {
-					ttyFd := int(os.Stdout.Fd())
+					ttyFd := int(os.Stdin.Fd())
```

Comment:
Hm, just to throw an edge-case your way - this should *also* work if `os.Stdin` is not really a tty. e.g. on linux this is like `echo "" | dagger core container from --address=alpine terminal`. Note, how this works - since stdin isn't a TTY, we try and open `/dev/tty` instead.

To handle this, we should use the same logic as we do with `findTTY`s https://github.com/dagger/dagger/blob/c91fe3d76dc580d662215f654b371b756f48bd1d/dagql/idtui/frontend_pretty.go#L206-L216 - perhaps we can save the result that we pass to `tea.WithInput` onto `frontendPretty` somewhere, and then use that instead?

Reply:
Hi @jedevc

do you have a specific scenario in mind where a user would do something like this?

Reply:
Yeah, it's more applicable when using `dagger shell`:

```
$ echo ".core | container | from alpine | terminal" | dagger shell
```

Then we have to spawn a `terminal`, but the stdin is actually just a pipe from `echo`.

Reply:
thank you Justin. 👍

I have pushed the code as suggested and also tested with Mac/Windows client.

thanks